common/xdc/nDCHistoryResender.go

```diff
@@ -128,13 +128,35 @@ func (n *NDCHistoryResenderImpl) SendSingleWorkflowHistory(
 
 		// we don't need to handle continue as new here
 		// because we only care about the current run
-		// TODO: revisit to evaluate if we need to handle continue as new
 		replicationRequest := n.createReplicationRawRequest(
 			domainID,
 			workflowID,
 			runID,
 			historyBatch.rawEventBatch,
 			historyBatch.versionHistory.GetItems())
+		lastEvent, err := n.getLastEvent(historyBatch.rawEventBatch)
```

Comment:
here is trying to decode every history event batch for the last event in that batch, try only decode the last batch

Reply:
good suggestion. done